spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/support/DefaultSubscriberFactory.java

```diff
@@ -39,9 +42,12 @@
 
 	private final CredentialsProvider credentialsProvider;
 
-	public DefaultSubscriberFactory(String projectId, ExecutorProvider executorProvider,
-			ChannelProvider channelProvider, CredentialsProvider credentialsProvider) {
-		this.projectId = projectId;
+	public DefaultSubscriberFactory(GcpProjectIdProvider projectIdProvider,
+			ExecutorProvider executorProvider,
+			ChannelProvider channelProvider,
+			CredentialsProvider credentialsProvider) {
+		this.projectId = projectIdProvider.getProjectId()
+				.orElseThrow(MissingProjectIdException::new);
 		this.executorProvider = executorProvider;
 		this.channelProvider = channelProvider;
 		this.credentialsProvider = credentialsProvider;
```

Comment:
I think all these options must be covered with the `Assert.notNull()` .
Also that `MissingProjectIdException`  should be replaced with an appropriate `Assert.notNull()` as well to follow with Spring standards for properties validations.